marlowe-playground-client/test/Marlowe/ContractTests.purs

```diff
@@ -20,91 +27,262 @@ import Examples.PureScript.EscrowWithCollateral as EscrowWithCollateral
+
+-- This is a property based test that checks that for a given contract, the possible actions available
+-- during the simulation don't throw errors nor warnings.
+contractDontHaveErrors :: String -> Term T.Contract -> TestSuite
+contractDontHaveErrors contractName contract =
+  test contractName
+    $ quickCheck
+    $ evalStateT property mkState
+  where
+  property :: StateT Simulation.State Gen Result
+  property = do
+    startSimulation zero contract
+    runContract (Just contract)
+
+  -- When the possible action is a Choose, we need to generate a random value inside the provided Bound.
+  -- We use this function (which looses precision) in order to select a small integer and wrap it in a BigInteger
+  looselyToInt :: BigInteger -> Int
+  looselyToInt = round <<< BigInteger.toNumber
+
+  genValueInBound :: forall m. MonadGen m => MonadRec m => Bound -> m BigInteger
+  genValueInBound (Bound from to) = BigInteger.fromInt <$> chooseInt (looselyToInt from) (looselyToInt to)
+
+  runContract :: Maybe (Term T.Contract) -> StateT Simulation.State Gen Result
+  runContract Nothing = pure Success
+
+  runContract (Just (Term T.Close _)) = pure Success
+
+  -- TODO: For the moment it was not needed, as none of the examples triggered a warning nor an error
+  --       but we should add an extra parameter to runContract that includes the current Action list
+  --       and in case of a failure, we should print what is the list that causes the problem.
+  runContract (Just _) = do
+    simulationState <- get
+    let
+      mPossibleActions :: Maybe (NonEmptyArray ActionInput)
+      mPossibleActions =
+        simulationState
+          # preview _currentPossibleActions
+          # map getAllActions
+          # bindFlipped fromArray
+    case mPossibleActions of
+      Nothing -> pure $ Failed "no available actions"
+      Just possibleActions -> do
+        action <- lift $ elements possibleActions
+        case action of
+          MoveToSlot slot -> moveToSlot slot
+          DepositInput accountId party token value -> applyInput $ IDeposit accountId party token value
+          ChoiceInput choiceId bounds value -> do
+            randomValue <- lift $ oneOf $ pure value :| (genValueInBound <$> bounds)
+            applyInput $ IChoice choiceId randomValue
+          NotifyInput -> applyInput INotify
+        (mError :: Maybe TransactionError) <- gets $ preview (_currentMarloweState <<< _executionState <<< _SimulationRunning <<< _transactionError <<< _Just)
+        (mWarnings :: Maybe (Array TransactionWarning)) <- gets $ preview (_currentMarloweState <<< _executionState <<< _SimulationRunning <<< _transactionWarnings)
+        mContract <- gets (preview _currentContract)
+        case mError, mWarnings of
+          Just err, _ -> pure $ Failed "it has errors"
+          _, Just warnings
+            | length warnings > 0 -> pure $ Failed "it has warnings"
```

Comment:
Would `Nothing` in `mWarnings` also be an error?

Reply:
So maybe it would be better to write something like:
```haskell
case mError, mWarnings of
  Nothing, Just [] -> runContract mContract
  -- Error analysis cases:
  Just err, _ -> pure $ Failed "It has errors" -- Maybe add error to the message
  _, Just warns -> pure $ Failed "It has warnings" -- Maybe add warnings to the message
  _, _ -> pure $ Failed "Wrong simulation state"
```
I may be confused here

Reply:
no, you are right. Much better :)